odoo/orm/fields_properties.py

```diff
@@ -29,6 +29,74 @@ def check_property_field_value_name(property_name):
     if not (0 < len(property_name) <= 512) or not regex_alphanumeric.match(property_name):
         raise ValueError(f"Wrong property field value name {property_name!r}.")
 
+class Property(abc.Mapping):
+    """An object that implements the value of a :class:`Properties` field in the "record"
+    format, i.e., the result of evaluating an expression like ``record.property_field``.
+    The value behaves as a ``dict``, and individual properties are returned in their
+    expected type, according to ORM conventions.  For instance, the value of a many2one
+    property is returned as a recordset::
+
+        # attributes is a properties field, and 'partner_id' is a many2one property;
+        # partner is thus a recordset
+        partner = record.attributes['partner_id']
+        partner.name
+
+    When the accessed key does not exist, i.e., there is no corresponding property
+    definition for that record, the access raises a :class:`KeyError`.
+    """
+
+    def __init__(self, values, field, record):
+        self._values = values or {}
+        self.record = record
+        self.field = field
+
+    def __len__(self):
+        return len(self._values)
+
+    def __iter__(self):
+        return iter(self._values)
+
+    def __getitem__(self, property_name):
+        """Will make the verification."""
+        if not self.record:
+            return False
+
+        records = self.record | self.record.browse(islice(self.record._prefetch_ids, 1000))
+        records = records.filtered(lambda r: r[self.field.definition_record] == self.record[self.field.definition_record])
+
+        values = self.field.convert_to_read(records.env.cache.get(records, self.field), self.record, use_display_name=False)
+        prop = next((p for p in values if p['name'] == property_name), False)
+        if not prop:
+            # We need to know if it's a relational property defined on a different definition record
```

Comment:
Please explain why "on a different definition record".  That part doesn't make any sense to me.  What's the point of defining a definition record if you can pick anyone in practice?

Reply:
will be dropped if you are ok with the solution "always raise if property doesn't exist, and safe expression in template"